pootle/static/js/editor/index.js

```diff
@@ -12,23 +12,25 @@ import React from 'react';
 import ReactRenderer from 'utils/ReactRenderer';
 import { q, qAll } from 'utils/dom';
 
-import Editor from './containers/Editor';
-import UnitSource from './components/UnitSource';
+import { loadFormatAdaptor } from './formats/FormatLoader';
 import { hasCRLF, normalize, denormalize } from './utils/normalizer';
 
 
 const ReactEditor = {
 
   init(props) {
-    this.node = q('.js-mount-editor');
-    this.sourceNode = q('.js-mount-editor-original-src');
-    this.alternativeSourceNodes = qAll('.js-mount-editor-alt-src');
-    this.props = {};
-    this.hasCRLF = props.sourceValues.some(hasCRLF);
-
-    ReactRenderer.unmountComponents();
-
-    this.setProps(props);
+    loadFormatAdaptor(props, (newProps, adaptor) => {
+      this.node = q('.js-mount-editor');
+      this.sourceNode = q('.js-mount-editor-original-src');
+      this.alternativeSourceNodes = qAll('.js-mount-editor-alt-src');
+      this.props = {};
+      this.hasCRLF = newProps.sourceValues.some(hasCRLF);
```

Comment:
I think we want to keep the assignments before this line (including this one) out of the callback function.